x/ccv/provider/keeper/grpc_query.go

```diff
@@ -232,18 +229,40 @@ func (k Keeper) QueryRegisteredConsumerRewardDenoms(goCtx context.Context, req *
 	}, nil
 }
 
-// TODO (PERMISSIONLESS)
 func (k Keeper) QueryProposedConsumerChainIDs(goCtx context.Context, req *types.QueryProposedChainIDsRequest) (*types.QueryProposedChainIDsResponse, error) {
```

Comment:
What's the use case for having this query? Without a clear use case, this query should be removed. All of this can be done on the frontend -- query all the proposals, filter only the ones still in voting and contain a MsgUpdateConsumer. The MsgUpdateConsumer will have the consumerId.

Reply:
Why was this query there in the first place? Do we know if someone is using it? I don't have the answers to these questions. The changes here were meant to keep the same behavior we had before. 
But if you do not see a use case, I'll go ahead and remove it.

Note that a multitude of things (e.g., [this one](https://github.com/cosmos/interchain-security/pull/2162)) could be done in the front end but nevertheless the front-end team asked us for exact queries (not this one though) to simplify their life.

Reply:
Comment it for now with a TODO and we handle it later in a discussion with the FE team.

Reply:
Or just remove it and we add it later if we need.